### PR TITLE
Update dockerfile, add requirements.txt and change dependabot

### PR DESCRIPTION
**Pull request by modem7**
Update dockerfile and add requirements.txt:
This allows for easier management and allows for dependabot to auto-open a PR when the pip packages are updated lowering management time.

Have also added the option for chmod of the entry.sh to be done during the copy stage rather than later, allowing this step to be cached when required, lowering build time and layer size.

This is more to initiate a discussion to see if it's suitable. If it is, msmtp and ntfy would need changing as well.

### CODE REVIEW

File: base/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+pip==22.1.1
+llfuse==1.4.1
+borgbackup==1.2.0
+borgmatic==1.6.0
```
Comment (grantbevis) [inline base/requirements.txt:4]: This needs to be 1.6.1 👍🏽

Comment (modem7) [reply base/requirements.txt]: > This needs to be 1.6.1 👍🏽

All updated!

**Comment by grantbevis** [comment]
Weird, there’s a merge conflict. I can resolve in the morning if you’re unable to

**Comment by modem7** [comment]
> Weird, there’s a merge conflict. I can resolve in the morning if you’re unable to

Odd. 

There we go, that should resolve.

**Review by grantbevis — APPROVED** [review]

**Comment by grantbevis** [comment]
Thanks @modem7

**Comment by modem7** [comment]
You got it!

If it's all working as expected, I'm happy to raise a PR for the other two versions as well.

Maybe give it 24-48 hours to see how things are affected/work.

**Pull request merged**